File: OOps/C++/enum.cpp
#include <iostream>
#include <vector>
using namespace std;

enum class OrderStatus
{
    PLACED,
    CONFIRMED,
    SHIPPED,
    DELIVERED,
    CANCELLED
};

enum class Coin
{
    PENNY,
    NICKEL,
    DIME,
    QUARTER
};

int getCoinValue(Coin coin)
{
    switch (coin)
    {
    case Coin::PENNY:
        return 1;
    case Coin::NICKEL:
        return 5;
    case Coin::DIME:
        return 10;
    case Coin::QUARTER:
        return 25;
    default:
        return 0;
    }
}

int main()
{

    OrderStatus status = OrderStatus::SHIPPED;

    if (status == OrderStatus::SHIPPED)
    {
        cout << "Your package is on the way!" << endl;
    }

    return 0;
}

File: OOps/Typescript/enum.ts
enum OrderStatus {
  PLACED,
  CONFIRMED,
  SHIPPED,
  DELIVERED,
  CANCELLED,
}

const shipped: OrderStatus = OrderStatus.SHIPPED;

if (shipped === OrderStatus.SHIPPED) {
  console.log("Your package is on the way!");
}

class Coin {
  static readonly PENNY = new Coin("PENNY", 1);
  static readonly NICKEL = new Coin("NICKEL", 5);
  static readonly DIME = new Coin("DIME", 10);
  static readonly QUARTER = new Coin("QUARTER", 25);

  private constructor(
    public readonly name: string,
    private readonly value: number,
  ) {}

  getValue(): number {
    return this.value;
  }
}

const total: number = Coin.PENNY.getValue() + Coin.DIME.getValue();

console.log(total)